### PR TITLE
Adds CCINIClass to TD and refactors scenario loading code to use it.

### LINKED ISSUE #210: [TD] Use the common INIClass
TD is using older version of ini management functions that are obsoleted by INIClass.

### PR DESCRIPTION
**Pull request by OmniBlade**
CCINIClass contains specialised routines for parsing certain data types from an INI file and was introduced in RA. This means the INI reading and writing code for object classes doesn't need to duplicate this code for certain types. By refactoring the scenario loading code to use the new TD version of this class, we bring the code more in line with RA code and lay the basis for more improvements such as rules.ini for TD.

### CODE REVIEW

File: tiberiandawn/ccini.cpp [hunk 1]
```diff
@@ -0,0 +1,1165 @@
...
+ *          default value is returned.                                                         *
+ *                                                                                             *
+ * WARNINGS:   none                                                                            *
+ *                                                                                             *
+ * HISTORY:                                                                                    *
+ *   07/03/1996 JLB : Created.                                                                 *
+ *=============================================================================================*/
+long CCINIClass::Get_Owners(char const* section, char const* entry, long defvalue) const
+{
+    return 0;
```
Comment (hifi) [inline tiberiandawn/ccini.cpp]: Would be better not to include stubs that are not used and return the incorrect data. When/if CCINIClass is merged to common we can add proper implementations and not ever accidentally use stubs.

So I would prefer to not have any functions in the class that are not used within TD but keep the same API for functions that are used by both games. This makes the class smaller and easier to grasp what's being actually used.

File: tiberiandawn/scenarioini.cpp [hunk 2]
```diff
@@ -272,42 +263,36 @@ bool Read_Scenario_Ini(char* root, bool fresh)
...
 
     /*
     ** Init the Scenario CRC value
     */
     ScenarioCRC = 0;
-    len = strlen(buffer);
-    for (int i = 0; i < len; i++) {
-        val = (unsigned char)buffer[i];
-#ifndef DEMO
-        Add_CRC(&ScenarioCRC, (unsigned long)val);
```
Comment (hifi) [inline tiberiandawn/scenarioini.cpp:296]: I didn't see this being implemented anywhere, doesn't this mean that the CRC is never calculated?

File: tiberiandawn/scenarioini.cpp [hunk 3]
```diff
@@ -715,48 +658,40 @@ bool Read_Scenario_Ini_File(char* scenario_file_name, char* bin_file_name, const
...
```
Comment (hifi) [inline tiberiandawn/scenarioini.cpp:745]: Ditto here.

**Review by hifi — CHANGES_REQUESTED** [review]
Aside from the CRC stuff which I didn't really understand I didn't find anything else sticking out. And the stubs.

You added some NULL checks along the way which is good. I haven't yet tested playing with these changes but I've painstakingly read through everything with as much thought I could maintain while doing that.

Overall good work, need to do some play testing to make sure these work as intended at least for a few scenario maps.

File: tiberiandawn/scenarioini.cpp (hunk 2 above)
Comment (OmniBlade) [reply tiberiandawn/scenarioini.cpp:296]: This is true, the CCINIClass uses the SHA1 digest instead which would need to be used in place of the CRC check which I assume is just to check no one tampered with the map in MP games.

File: tiberiandawn/ccini.cpp (hunk 1 above)
Comment (OmniBlade) [reply tiberiandawn/ccini.cpp]: I've removed the implementations, do you want me to remove the prototypes as well?

File: tiberiandawn/scenarioini.cpp (hunk 2 above)
Comment (hifi) [reply tiberiandawn/scenarioini.cpp:296]: Would it work to fill the crc with a partial from the SHA digest so it at the least functions for its intended purpose? We can then replace it with the full digest once we touch the protocol.

File: tiberiandawn/ccini.cpp (hunk 1 above)
Comment (hifi) [reply tiberiandawn/ccini.cpp]: That would be great as well. Thanks.

**Review by hifi — APPROVED** [review]
I tested the game runs with these changes. Looks good enough to me now.

**Pull request merged**